src/tasks/tasks.hpp

```diff
@@ -120,6 +131,9 @@ class Task {
 
   TaskStatus operator()() {
     auto status = f();
+    if (verbose_level_ > 0)
+      std::cout << label_ << " [status = " << static_cast<int>(status) << "]"
+                << std::endl;
```

Comment:
Should these include MPI rank? Also `printf` more than stream is probably better because with MPI streams can get muddled. Better to send a whole string to `cout` at once.

Reply:
sure, that is a valid concern. I will include rank as well.

Reply:
done